packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -35,22 +37,117 @@ import {validateOptions} from '../../../utils/validate-payload';
 import {
   CategoryFacetOptions,
   categoryFacetOptionsSchema,
+  CategoryFacetSearchOptions,
 } from './headless-category-facet-options';
 import {determineFacetId} from '../_common/facet-id-determinor';
+import {FacetSearch} from '../facet/headless-facet';
+import {
+  BaseFacetSearchResult,
+  BaseFacetSearchState,
+  BaseFacetState,
+  BaseFacetValue,
+} from '../_common/base-facet';
 
-export {CategoryFacetOptions};
-export type CategoryFacetProps = {
+export {CategoryFacetOptions, CategoryFacetSearchOptions};
+
+export interface CategoryFacetProps {
   /** The options for the `CategoryFacet` controller. */
   options: CategoryFacetOptions;
-};
+}
+
 /**
  * The `CategoryFacet` headless controller offers a high-level interface for designing a facet UI controller that renders values in a hierarchical fashion.
  */
-export type CategoryFacet = ReturnType<typeof buildCategoryFacet>;
+export interface CategoryFacet extends Controller {
+  facetSearch: CategoryFacetSearch;
+
+  /**
+   * Toggles the specified facet value.
+   * @param selection The facet value to toggle.
+   */
+  toggleSelect: (selection: CategoryFacetValue) => void;
+
+  /** Deselects all facet values.*/
+  deselectAll: () => void;
+
+  /** Sorts the facet values according to the specified criterion.
+   * @param criterion The criterion to sort values by.
+   */
+  sortBy: (criterion: CategoryFacetSortCriterion) => void;
+
+  /**
+   * Checks whether the facet values are sorted according to the specified criterion.
+   * @param criterion The criterion to compare.
+   * @returns Whether the facet values are sorted according to the specified criterion.
+   */
+  isSortedBy: (criterion: CategoryFacetSortCriterion) => boolean;
+
+  /**
+   * Increases the number of values displayed in the facet to the next multiple of the originally configured value.
+   */
+  showMoreValues: () => void;
+
+  /** Sets the displayed number of values to the originally configured value. */
+  showLessValues: () => void;
+
+  state: CategoryFacetState;
+}
+
 /**
  * A scoped and simplified part of the headless state that is relevant to the `CategoryFacet` controller.
  */
-export type CategoryFacetState = CategoryFacet['state'];
+export interface CategoryFacetState extends BaseFacetState {
+  /** The facet id. */
+  facetId: string;
+
+  /** The parent values of the facet. */
+  parents: CategoryFacetValue[];
+
+  /** The values of the facet. */
+  values: CategoryFacetValue[];
+
+  /** `true` if there are more values to display and `false` otherwise. */
+  canShowMoreValues: boolean | undefined;
```

Comment:
Because of [this API member](https://github.com/coveo/ui-kit/blob/0060591505cf22c34fc9ca4e467661186a14c940/packages/headless/src/features/facets/category-facet-set/interfaces/response.ts#L7), this can be `undefined`. Should I assert that it's not undefined?

Reply:
Yes, I would prefer for the type to just be a boolean. Are you sure the internal type is correct and can actually be `undefined`? If not, correcting the internal interface would be best.

If it can be `undefined`, then inside the `state` getter let's assert and return `false` when it is `undefined`?

Reply:
I'll see or ask around whether this can really be undefined, it's weird.